### PR TITLE
Added MapPanel & MapPanelDrawer components

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: src/components/MapDrawer/constants.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Position of the drawer when it's at the bottom
+export const DRAWER_HANDLE_HEIGHT = 50
+export const DRAWER_HANDLE_PADDING = 50
+export const SNAP_OFFSET = 0.5 // 0 - 1, increase this to snap quicker to bottom or top, instead of middle
+export const DRAWER_HANDLE_PEEK = 20
+
+export enum SnapPoint {
+  Top = 0,
+  Middle = window.innerHeight / 2,
+  Bottom = window.innerHeight - DRAWER_HANDLE_HEIGHT - DRAWER_HANDLE_PEEK,
```
Comment (timvanoostrom) [inline src/components/MapDrawer/constants.ts:10]: This won't survive a resize right?

Comment (timvanoostrom) [reply src/components/MapDrawer/constants.ts:10]: Sorry if I'm too early :-)

File: example/src/pages/MapBaseLayers.tsx [hunk 2]
```diff
@@ -55,14 +48,7 @@ const MapBaseLayers: React.FC = () => {
...
   return (
-    <StyledMap options={DEFAULT_AMSTERDAM_MAPS_OPTIONS}>
-      <Scale
-        options={{
-          position: 'bottomright',
-          metric: true,
-          imperial: false,
-        }}
-      />
+    <Map fullScreen>
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply example/src/pages/MapBaseLayers.tsx:51]: This one should be fixed in https://github.com/Amsterdam/amsterdam-react-maps/pull/253

File: example/src/pages/MapPanelPage.tsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+  useEffect(() => {
+    if (matchPositionWithSnapPoint(SnapPoint.Halfway) && currentLatLng) {
+      pan(currentLatLng, variant === 'drawer' ? 'vertical' : 'horizontal', 20)
+    }
+  }, [drawerPosition, currentLatLng])
+  return currentLatLng ? <Marker latLng={currentLatLng} /> : null
+}
+
+const MapLegendContent = ({ ...otherProps }) => (
+  <MapPanelContent title="Legenda" {...otherProps}>
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply example/src/pages/MapPanelPage.tsx:140]: Ah this is in FF, fixed it now :)

File: example/src/pages/MapPanelPage.tsx [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+            />
+          }
+          bottomRight={<Controls />}
+        />
+      )}
+    </>
+  )
+}
+
+const GlobalStyle = createGlobalStyle`
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply example/src/pages/MapPanelPage.tsx:94]: not sure, should we? don't know what side-effects this would cause

**Pull request merged**